sources/packages/api/src/route-controllers/designation-agreement/designation-agreement.aest.controller.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  Controller,
+  Get,
+  Param,
+  NotFoundException,
+  Patch,
+  Body,
+  Query,
+  BadRequestException,
+} from "@nestjs/common";
+import { DesignationAgreementService, FormService } from "../../services";
+import { FormNames } from "../../services/form/constants";
+import { DesignationAgreementStatus } from "../../database/entities";
+import { getISODateOnlyString, PaginationParams } from "../../utilities";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { AllowAuthorizedParty, Groups, UserToken } from "../../auth/decorators";
+import { IUserToken } from "../../auth/userToken.interface";
+import { UserGroups } from "../../auth/user-groups.enum";
+import {
+  GetDesignationAgreementDto,
+  GetDesignationAgreementsDto,
+  PendingDesignationDto,
+  UpdateDesignationDto,
+} from "./models/designation-agreement.model";
+import { DesignationAgreementServiceController } from "./designation-agreement.service.controller";
+
+@AllowAuthorizedParty(AuthorizedParties.aest)
+@Groups(UserGroups.AESTUser)
+@Controller("designation-agreement")
+export class DesignationAgreementAESTController {
+  constructor(
+    private readonly designationAgreementServiceController: DesignationAgreementServiceController,
+    private readonly designationAgreementService: DesignationAgreementService,
+    private readonly formService: FormService,
+  ) {}
+
+  /**
+   * Retrieve the designation agreement information and
+   * the associated locations approvals.
+   * @param designationId
+   * @returns designation agreement information.
+   */
+  @Get(":designationId")
+  async getDesignationAgreement(
+    @Param("designationId") designationId: number,
+  ): Promise<GetDesignationAgreementDto> {
+    return this.designationAgreementServiceController.getDesignationAgreement(
+      designationId,
+    );
+  }
+
+  /**
+   * Get the list of all the designations that belongs to
+   * the institution.
+   * @param institutionId
+   * @returns the list of all the designations that
+   * belongs to the institution.
+   */
+  @Get("institution/:institutionId")
+  async getDesignationAgreements(
+    @Param("institutionId") institutionId: number,
+  ): Promise<GetDesignationAgreementsDto[]> {
+    return this.designationAgreementServiceController.getDesignationAgreements(
+      institutionId,
+    );
+  }
+
+  /**
+   * API to retrieve all pending designations.
+   * @param designationStatus
+   * @param searchCriteria to search designation.
+   * @returns Pending designations.
+   */
+  @Get("status/:designationStatus")
+  async getDesignationAgreementByStatus(
+    @Param("designationStatus") designationStatus: DesignationAgreementStatus,
+    @Query(PaginationParams.SearchCriteria) searchCriteria: string,
+  ): Promise<PendingDesignationDto[]> {
+    if (
+      !Object.values(DesignationAgreementStatus).includes(designationStatus)
```

Comment:
Does this validation needed? as we are passing the type of the designationStatus as enum of DesignationAgreementStatus in the param?

Reply:
@guru-aot, at runtime Nest will not enforce the enum validation (it is really just a string). The only way to Nest enforce it is when the enum belongs to the payload (not the URI) and we can use the class validators.